tests/main/classic-ubuntu-core-transition-two-cores/task.yaml

```diff
@@ -6,6 +6,15 @@ systems: [-ubuntu-core-16-*, -ubuntu-*-ppc64el]
 
 warn-timeout: 1m
 kill-timeout: 5m
+restore: |
+    rm -f state.json.new
+    systemctl stop snapd.{service,socket} || true
+    # remove any generated files and force system key re-generate
+    rm -f /etc/apparmor.d/snap.core.*.usr.lib.snapd.snap-confine*
+    rm -f /etc/apparmor.d/cache/*
+    rm -f /var/lib/snapd/system-key
```

Comment:
I think we should never end up in a situation where we have to regenerate system key. It contains build id and all kinds of other stuff. I don't understand why we'd need this code.

Reply:
The /etc/apparmor.d/snap.core.*.snap-confine profiles are only generated when the system key changes. I suspect that something during the tests generates a different snap.core.*.snap-confine profile. Then we restore our test profile during the restore phase of the test suite. However this will not lead to re-generation of the profile we need nor does it lead to re-loading this profile (because the system key has not changed). So yes, we should never need this except that during tests we do all sorts of things behind the back of snapd.